### PR TITLE
Add Windows Firewall settings + Restart UAC

### PR DESCRIPTION
**Pull request by SadPencil**
No one likes the Windows Firewall pop-up window.
This pull request implement a new feature to check and set up firewall settings to avoid the firewall pop-up window.
If the firewall settings are there, nothing happens.
Otherwise, the client will ask for user would he/she want to restart the client and gain administrator privileges. Once user clicked 'yes', the client will restart and set up firewall settings.
XNAMessagebox is used to ask user.

A new key in `ClientDefination.ini` is set as `BypassWindowsFirewall` with default value `true`.

Tested on DTA and MO.

### CODE REVIEW

**Comment by Metadorius** [comment]
A screenshot of the window would be handy :wink:

**Comment by SadPencil** [comment]
@Metadorius 
https://i.postimg.cc/xCLLbwZ4/001.png
https://i.postimg.cc/tJ2Z3pN2/002.png

**Comment by SadPencil** [comment]
Update:
If two XNAMessageBoxes are shown, i.e. the first run messagebox and firewall messagebox at the same time, there is chaos.
So currently, fallback to the common MessageBox which shows and blocks the thread.

File: DXMainClient/DXGUI/Generic/MainMenu.cs [hunk 1]
```diff
@@ -351,28 +352,26 @@ private void CheckIfFirstRun()
...
+                xnaMessageBox = XNAMessageBox.ShowYesNoDialog(WindowManager, "Initial Installation",
                     string.Format("You have just installed {0}." + Environment.NewLine +
                     "It's highly recommended that you configure your settings before playing." +
                     Environment.NewLine + "Do you want to configure them now?", ClientConfiguration.Instance.LocalGame));
-                firstRunMessageBox.YesClickedAction = FirstRunMessageBox_YesClicked;
-                firstRunMessageBox.NoClickedAction = FirstRunMessageBox_NoClicked;
+                xnaMessageBox.YesClickedAction = (XNAMessageBox messageBox) =>
+                {
+                    optionsWindow.Open();
+                };
```
Comment (Rampastring) [inline DXMainClient/DXGUI/Generic/MainMenu.cs]: No need to have brackets in a single-line lambda function.

**Comment by Rampastring** [comment]
Tbh I don't think it's the client's job to change firewall permissions.

DTA already shows the compatibility fix and first run dialog at the same time on the first run on a new system, and it works okay.

**Comment by SadPencil** [comment]
Yes. I don't know where the dialog should be put on. 

It's true that the client should not change firewall settings. I just find the two Firewall pop-up windows are annoying, one of which players can only pause the game, return to Windows and deal with the Firewall pop-up window. So client asking the user's permission and then set-up Windows Firewall can reduce this annoying thing.

However, my pull request is indeed not an elegant way because it creates a new in-game pop-up window. I don't know what's the best way.

**Comment by SadPencil** [comment]
There are more optimizations can be done. See https://github.com/CnCNet/xna-cncnet-client/issues/132

**Pull request closed without merging**